src/pow.cpp

```diff
@@ -428,6 +432,8 @@ RCP<const Basic> pow_expand(const RCP<const Pow> &self)
         }
     }
     RCP<const Basic> result = Add::from_dict(add_overall_coeff, std::move(rd));
+    RCP<const Number> i = integer(-1);
+    if(negative_pow) result = rcp(new Pow(result, i));
```

Comment:
I see, that's right. The other option is to put an assert statement above the `pow` to check that the `result` is `Add`. Then I think it's safe and faster.